### PR TITLE
Drop Python2 and Six

### PR DESCRIPTION
**Pull request by oz123**

### CODE REVIEW

File: src/vistir/contextmanagers.py [hunk 1]
```diff
@@ -345,7 +344,8 @@ def open_file(
             else:
                 session = Session()
         if session is None:
-            with closing(urlopen(link)) as f:
+            import urllib.request
```
Comment (matteius) [inline src/vistir/contextmanagers.py:347]: This style of import is problematic for pipenv to rewrite -- I converted `requirementslib` off this style imports for that reason, can you do `from urllib import request`?  Also why not have it at the top of the file?

**Comment by matteius** [comment]
Looks like two editor .swp files got added.  Also this would be a good chance to fix the import issues/catching ModuleNotfound error that caused some issues with removing yaspin:  https://github.com/pypa/pipenv/pull/5268

File: src/vistir/contextmanagers.py (hunk 1 above)
Comment (matteius) [reply src/vistir/contextmanagers.py:347]: may not matter for this though since its a python builtin library.

Comment (oz123) [reply src/vistir/contextmanagers.py]: I fixed this as requested.

**Comment by oz123** [comment]
> Looks like two editor .swp files got added. Also this would be a good chance to fix the import issues/catching ModuleNotfound error that caused some issues with removing yaspin: [pypa/pipenv#5268](https://github.com/pypa/pipenv/pull/5268)

Removed the `swp` files from vim, also added these to git ignore.

As for the issue with yaspin, I can't reproduce it. However, pipenv does not even use yaspin, and vistir has yaspin only as an optional dependency. Hence, it should be safe to remove them from pipenv.

**Comment by matteius** [comment]
> As for the issue with yaspin

@oz123 -- basically because in your PR the module folder itself gets deleted, some pythons will apparently raise `ModuleNotFoundError` instead of `ImportError` in this case, so this line should change to catch both error types and it should become safe for all:  https://github.com/sarugaku/vistir/blob/master/src/vistir/contextmanagers.py#L180

**Review by matteius — APPROVED** [review]

**Pull request merged**